coalib/bearlib/abstractions/Linter.py

```diff
@@ -565,21 +591,82 @@ def run(self, filename, file, **kwargs):
+            """
+            Creates the arguments for the linter.
 
+            You can provide additional keyword arguments and defaults. These
+            will be interpreted as required settings that need to be provided
+            through a coafile-section.
+
+            :param config_file:
+                The path of the config-file if used. ``None`` if unused.
+            :return:
+                A sequence of arguments to feed the linter-tool with.
+            """
+            raise NotImplementedError
+
+        def run(self, **kwargs):
+            # Get the **kwargs params to forward to `generate_config()`
+            # (from `_create_config()`).
+            generate_config_kwargs = FunctionMetadata.filter_parameters(
+                self._get_generate_config_metadata(), kwargs)
+
+            with self._create_config(
+                    **generate_config_kwargs) as config_file:
+                # And now retrieve the **kwargs for `create_arguments()`.
+                create_arguments_kwargs = (
+                    FunctionMetadata.filter_parameters(
+                        self._get_create_arguments_metadata(), kwargs))
+                args = self.create_arguments(config_file,
+                                             **create_arguments_kwargs)
+
+                try:
+                    args = tuple(args)
+                except TypeError:
+                    self.err('The given arguments '
+                             '{!r} are not iterable.'.format(args))
+                    return
+
+                arguments = (self.get_executable(),) + args
+                self.debug("Running '{}'".format(' '.join(arguments)))
+
+                output = run_shell_command(
+                    arguments,
+                    stdin=None,
+                    cwd=self.get_config_dir())
+
+                output = tuple(compress(
+                    output,
+                    (options['use_stdout'], options['use_stderr'])))
+                if len(output) == 1:
+                    output = output[0]
+
+                process_output_kwargs = FunctionMetadata.filter_parameters(
+                    self._get_process_output_metadata(), kwargs)
+                return self.process_output(output, **process_output_kwargs)
+
+    if options['global_bear']:
+        LinterBaseClass = GlobalLinterBase
+    else:
+        LinterBaseClass = LocalLinterBase
     # Mixin the linter into the user-defined interface, otherwise
     # `create_arguments` and other methods would be overridden by the
     # default version.
     result_klass = type(klass.__name__, (klass, LinterBase), {
         '__module__': klass.__module__})
+    result_klass = type(klass.__name__, (klass, LinterBaseClass), {
+        '__module__': klass.__module__})
     result_klass.__doc__ = klass.__doc__ or ''
     return result_klass
 
 
 @enforce_signature
 def linter(executable: str,
+           global_bear: bool=False,
            use_stdin: bool=False,
```

Comment:
you'll probably want to document and catch the case global_bear=True and use_stdin=True which is incompatible.